modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -119,6 +122,9 @@ GrainTracker::finalize()
                                                 _unique_grain_to_ebsd_num[grain_pair.first] :
                                                 grain_pair.first,
                                                 grain_pair.second._var_idx);
+
+          auto elem_it_pair = _elemental_data_2.emplace(elem_id, std::vector<unsigned int>(_n_vars, libMesh::invalid_uint));
+          elem_it_pair.first->second[grain_pair.second._var_idx] = _ebsd_reader ? _unique_grain_to_ebsd_num[grain_pair.first] : grain_pair.first;
```

Comment:
Wait - but in the case that this element is already in the map you're going to allocate a `std::vector` of size `_n_vars` and then throw it away (it's a temporary that will go out of scope).

That's a LOT of memory churn that could be avoided by checking the map for the key first (which I know isn't as elegant).

The other way to avoid that is to not size the `std::vector` in your `emplace` call... and instead... size it after you get the iterator out.

The first time through it will be resized... every other time that resize will be a no-op.